### PR TITLE
Support TOTP two-factor authentication

### PR DESCRIPTION
**Pull request by JohnNiang**
#### What type of PR is this?

/kind feature
/area core

#### What this PR does / why we need it:

See https://github.com/halo-dev/halo/issues/4541 for more.

#### Which issue(s) this PR fixes:

Fixes https://github.com/halo-dev/halo/issues/4541

#### Does this PR introduce a user-facing change?

```release-note
支持 2FA 认证方式
```

### CODE REVIEW

**Comment by JohnNiang** [comment]
Hi @ruibaby ，似乎缺失 2FA 页面的 UI。

**Comment by ruibaby** [comment]
> Hi @ruibaby ，似乎缺失 2FA 页面的 UI。

想想咋整，如果用页面提供，会影响原来登录失效弹框的流程。

**Comment by JohnNiang** [comment]
> > Hi @ruibaby ，似乎缺失 2FA 页面的 UI。
> 
> 想想咋整，如果用页面提供，会影响原来登录失效弹框的流程。

暂时不考虑页面提供吧。按照目前提供的登录页面和登录弹窗单独适配即可。

**Comment by JohnNiang** [comment]
Hi @ruibaby , I find it difficult to resolve conflicts involved in the console, please help me resolve them before continuing. ❤️

**Comment by ruibaby** [comment]
> Hi @ruibaby , I find it difficult to resolve conflicts involved in the console, please help me resolve them before continuing. ❤️ 

Okay，I will reset and then place it in the user center.

**Comment by ruibaby** [comment]
/retitle Support TOTP two-factor authentication 

Ready for review

**Review by ruibaby — COMMENTED** [review]
/lgtm

File: application/src/main/java/run/halo/app/security/authentication/twofactor/totp/DefaultTotpAuthService.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        this.encryptor = loadOrCreateEncryptor(keysRoot);
+    }
+
+    private BytesEncryptor loadOrCreateEncryptor(Path keysRoot) {
+        try {
+            if (Files.notExists(keysRoot)) {
+                Files.createDirectories(keysRoot);
+            }
+            var keyStorePath = keysRoot.resolve("halo.keystore");
+            var password = "changeit".toCharArray();
```
Comment (guqing) [inline application/src/main/java/run/halo/app/security/authentication/twofactor/totp/DefaultTotpAuthService.java:46]: - [ ] 出于安全性考虑，建议为此提供一个配置项来让用户可以指定 keystore 的密码

**Review by guqing — APPROVED** [review]

**Pull request merged**